### DISCUSSION TITLE
How to use unsigned integer?

### DISCUSSION THREAD

**Discussion by azzamsa** [async-graphql/async-graphql#1128] [Q&A] [answered]
Hi.

> Since any value going over the wire is eventually transformed into JSON, you're also limited in the data types you can use.

What is the recommended way for unsigned integer?

Do you use 1) String, then convert it to integer, 2) [StringNumber in async\_graphql::types - Rust](https://docs.rs/async-graphql/latest/async_graphql/types/struct.StringNumber.html), or is there any better approach?

**Comment by xamgore** [comment, marked as answer]
I believe it should be a custom scalar, like `UInt`. Here is a [doc section](https://async-graphql.github.io/async-graphql/en/custom_scalars.html) about it.

**Reply by azzamsa** [reply]
Thanks!

**Reply by sunli829** [reply]
You can use `StringNumber` directly

**Reply by azzamsa** [reply]
> You can use `StringNumber` directly

Oh, I see. Thanks! [Custom scalars - Async-graphql Book](https://async-graphql.github.io/async-graphql/en/custom_scalars.html#custom-scalars)
